### PR TITLE
[browser] detect and assert engine features

### PR DESCRIPTION
**Pull request by pavelsavara**
This will fail fast during startup, if we don't have necessary JS APIs.

Contributes to https://github.com/dotnet/runtime/issues/84574
Contributes to https://github.com/dotnet/runtime/issues/76961

### CODE REVIEW

**Review by kg — APPROVED** [review]

File: src/mono/wasm/runtime/web-socket.ts [hunk 1]
```diff
@@ -26,7 +26,20 @@ let mono_wasm_web_socket_close_warning = false;
 const ws_send_buffer_blocking_threshold = 65536;
 const emptyBuffer = new Uint8Array();
 
+function verifyEnvironment() {
+    if (ENVIRONMENT_IS_SHELL) {
```
Comment (maraf) [inline src/mono/wasm/runtime/web-socket.ts:30]: Shoud we move it after the `typeof globalThis.WebSocket !== "function"` condition?

Comment (pavelsavara) [reply src/mono/wasm/runtime/web-socket.ts:30]: for some random polyfill ? I don't think so.

Comment (pavelsavara) [reply src/mono/wasm/runtime/web-socket.ts:30]: It has complex dynamic behavior and even the nodeJS polyfill doen't work well in all use-cases. I'm just loud and clear this is not supported.

**Pull request merged**